### DISCUSSION TITLE
hacking-together-an-asciimath-parser-for-dotnet/

### DISCUSSION THREAD

**Discussion by giscus** [andrewlock/blog-comments#417] [Announcements]
# hacking-together-an-asciimath-parser-for-dotnet/

In this post I describe how I hacked together an AsciiMath parser for .NET. I also show the existing implementations available and how I chose between them

https://andrewlock.net/hacking-together-an-asciimath-parser-for-dotnet/

**Comment by WorldMaker** [comment]
Maybe a fun aside, but in your research on AsciiMath did you bump into UnicodeMath? https://unicodemath.org/

I think UnicodeMath is neat because I find it far more readable than AsciiMath. (It looks closer to the final formatted MathML.) It’s also neat that Microsoft Office applications support UnicodeMath (and Math subset of LaTeX these days, but not AsciiMath). It’s obvious downside is harder to type, you need to be very familiar with your OS’ unicode supplemental keyboards (emoji keyboards), but as the classic Python adage goes “code [or in this case math] is read a lot more than it is written”.
